data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSourceBufferAccumulator.java

```diff
@@ -0,0 +1,151 @@
+    private final Counter kafkaConsumerWriteError;
+    private static final String KAFKA_CONSUMER_BUFFER_WRITE_ERROR = "kafkaConsumerBufferWriteError";
+    private static final int MAX_FLUSH_RETRIES_ON_IO_EXCEPTION = Integer.MAX_VALUE;
+    private static final Duration INITIAL_FLUSH_RETRY_DELAY_ON_IO_EXCEPTION = Duration.ofSeconds(5);
+    private final JsonFactory jsonFactory = new JsonFactory();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public KafkaSourceBufferAccumulator(final TopicsConfig topicConfigs,
+                                        final KafkaSourceConfig kafkaSourceConfig,
+                                        final String schemaType, PluginMetrics pluginMetric) {
+        this.kafkaSourceConfig = kafkaSourceConfig;
+        this.topicConfig = topicConfigs;
+        this.schemaType = schemaType;
+        this.pluginMetrics = pluginMetric;
+        this.kafkaConsumerWriteError = pluginMetrics.counter(KAFKA_CONSUMER_BUFFER_WRITE_ERROR);
+    }
+
+    public Record<Object> getEventRecord(final String line) {
+        Map<String, Object> message = new HashMap<>();
+        if (MessageFormat.getByMessageFormatByName(schemaType)
+                .equals(MessageFormat.PLAINTEXT)) {
+            message.put(MESSAGE_KEY, line);
+        } else if (MessageFormat.getByMessageFormatByName(schemaType)
+                .equals(MessageFormat.JSON)) {
+            try {
+                final JsonParser jsonParser = jsonFactory.createParser(line);
+                message = objectMapper.readValue(jsonParser, Map.class);
+            } catch (Exception e) {
+                LOG.error("Unable to parse json data [{}], assuming plain text", line, e);
+                message.put(MESSAGE_KEY, line);
+            }
+
+        }
+        Event event = JacksonLog.builder().withData(message).build();
+        return new Record<>(event);
+    }
+
+    public void write(List<Record<Object>> kafkaRecords, final Buffer<Record<Object>> buffer) throws Exception {
+        try {
+            writeAllRecordToBuffer(kafkaRecords,
+                    buffer, topicConfig);
+            LOG.info("Total number of records publish in buffer {} for Topic : {}", kafkaRecords.size(), topicConfig.getName());
+        } catch (Exception e) {
+            if (canRetry(e)) {
+                writeWithBackoff(kafkaRecords, buffer, topicConfig);
+            }
+            LOG.error("Error occurred while writing data to the buffer {}", e.getMessage());
+            kafkaConsumerWriteError.increment();
+        }
+    }
+
+    public synchronized void writeAllRecordToBuffer(List<Record<Object>> kafkaRecords, final Buffer<Record<Object>> buffer, final TopicsConfig topicConfig) throws Exception {
+        buffer.writeAll(kafkaRecords,
+                topicConfig.getBufferDefaultTimeout().toSecondsPart());
+        LOG.info("Total number of records publish in buffer {} for Topic : {}", kafkaRecords.size(), topicConfig.getName());
```

Comment:
same as above, should be removed to avoid flooding logs.

Reply:
Done.